devlib/module/thermal.py

```diff
@@ -68,7 +70,25 @@ def set_enabled(self, enabled=True):
     def get_temperature(self):
         """Returns the temperature of the thermal zone"""
         temp_file = self.target.path.join(self.path, 'temp')
-        return self.target.read_int(temp_file)
+        try:
+            temp_value = self.target.read_int(temp_file)
+        except CalledProcessError:
+            temp_value = -1
```

Comment:
Unless strictly necessary we really don't want in-band signaling like that. If the file does not exist or is unreadable, that function should raise an exception. This way the caller will never be at risk of manipulating a meaningless value.